File: backend/internal/types/task.go
package types

type Task struct {
	ID               string  `json:"id"`
	CollectionID     string  `json:"collection_id"`
	NotionDatabaseID string  `json:"notion_database_id,omitempty"`
	Title            string  `json:"title"`
	Status           string  `json:"status"` // backlog, this_week, today, done
	EstimatedTime    int     `json:"estimated_time"` // minutes
	ActualTime       int     `json:"actual_time"` // minutes (cumulative)
	Source           string  `json:"source"` // always "notion"
	NotionPageID     string  `json:"notion_page_id"`
	CompletedAt      *int64  `json:"completed_at,omitempty"` // unix timestamp
	CreatedAt        int64   `json:"created_at"`
	SortOrder        int     `json:"sort_order"`
}

File: packages/shared/src/types/task.ts
/** Status values returned by the Linkdo task API. */
export type TaskStatus = "backlog" | "this_week" | "today" | "done";

/** Shared task representation used by the Desktop and Admin applications. */
export interface ITask {
  uuid: string;
  collection_uuid: string;
  notion_database_uuid?: string;
  clickup_list_uuid?: string;
  title: string;
  content: string;
  status: TaskStatus;
  initial_status?: TaskStatus;
  notion_page_id?: string;
  clickup_task_id?: string;
  estimated_time: number;
  actual_time: number;
  scheduled_date?: string;
  completed_at?: string;
  sort_order?: string;
  created_at: string;
  updated_at: string;
}

export interface ITimeSession {
  uuid: string;
  task_uuid: string;
  started_at: string;
  ended_at?: string;
  duration: number;
}
